### PR TITLE
Refactor migration inventory finished percentage

### PR DESCRIPTION
**Pull request by azexcy**
Changes proposed in this pull request:
  - Refactor migration inventory finished percentage
  - Add unit test

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by azexcy** [comment]
```
mysql> show migration status 'j0101395cd93b2cfc189f29958b8a0342e882';
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| item | data_source | status                 | active | processed_records_count | inventory_finished_percentage | incremental_idle_seconds | error_message |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| 0    | sysbench_ds | EXECUTE_INVENTORY_TASK | true   | 1812000                 | 9                             |                          |               |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
1 row in set (0.02 sec)

mysql> show migration status 'j0101395cd93b2cfc189f29958b8a0342e882';
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| item | data_source | status                 | active | processed_records_count | inventory_finished_percentage | incremental_idle_seconds | error_message |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| 0    | sysbench_ds | EXECUTE_INVENTORY_TASK | true   | 10864000                | 54                            |                          |               |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
1 row in set (0.17 sec)

mysql> show migration status 'j0101395cd93b2cfc189f29958b8a0342e882';
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| item | data_source | status                 | active | processed_records_count | inventory_finished_percentage | incremental_idle_seconds | error_message |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| 0    | sysbench_ds | EXECUTE_INVENTORY_TASK | true   | 19996000                | 99                            |                          |               |
+------+-------------+------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
1 row in set (0.05 sec)

mysql> show migration status 'j0101395cd93b2cfc189f29958b8a0342e882';
+------+-------------+--------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| item | data_source | status                   | active | processed_records_count | inventory_finished_percentage | incremental_idle_seconds | error_message |
+------+-------------+--------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
| 0    | sysbench_ds | EXECUTE_INCREMENTAL_TASK | true   | 20000000                | 100                           |                          |               |
+------+-------------+--------------------------+--------+-------------------------+-------------------------------+--------------------------+---------------+
1 row in set (0.03 sec)
```

File: features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/migration/distsql/handler/query/ShowMigrationJobStatusQueryResultSet.java [hunk 1]
```diff
@@ -45,24 +45,19 @@ public final class ShowMigrationJobStatusQueryResultSet implements DatabaseDistS
...
     public void init(final ShardingSphereDatabase database, final SQLStatement sqlStatement) {
-        long currentTimeMillis = System.currentTimeMillis();
         List<InventoryIncrementalJobItemInfo> jobItemInfos = JOB_API.getJobItemInfos(((ShowMigrationStatusStatement) sqlStatement).getJobId());
         data = jobItemInfos.stream().map(each -> {
+            MigrationJobItemInfo jobItemInfo = (MigrationJobItemInfo) each;
             Collection<Object> result = new LinkedList<>();
-            result.add(each.getShardingItem());
-            InventoryIncrementalJobItemProgress jobItemProgress = each.getJobItemProgress();
+            result.add(jobItemInfo.getShardingItem());
+            InventoryIncrementalJobItemProgress jobItemProgress = jobItemInfo.getJobItemProgress();
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/migration/distsql/handler/query/ShowMigrationJobStatusQueryResultSet.java:53]: It's better to keep use `each` for common methods, it's enough

File: features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/migration/distsql/handler/query/ShowMigrationJobStatusQueryResultSet.java [hunk 2]
```diff
@@ -45,24 +45,19 @@ public final class ShowMigrationJobStatusQueryResultSet implements DatabaseDistS
...
                 result.add(jobItemProgress.getProcessedRecordsCount());
-                result.add(jobItemProgress.getInventory().getInventoryFinishedPercentage());
-                String incrementalIdleSeconds = "";
-                if (jobItemProgress.getIncremental().getIncrementalLatestActiveTimeMillis() > 0) {
-                    long latestActiveTimeMillis = Math.max(each.getStartTimeMillis(), jobItemProgress.getIncremental().getIncrementalLatestActiveTimeMillis());
-                    incrementalIdleSeconds = String.valueOf(TimeUnit.MILLISECONDS.toSeconds(currentTimeMillis - latestActiveTimeMillis));
-                }
-                result.add(incrementalIdleSeconds);
+                result.add(jobItemInfo.getInventoryFinishedPercentage());
+                result.add(jobItemInfo.getIncrementalIdleSeconds());
```
Comment (sandynz) [inline features/sharding/distsql/handler/src/main/java/org/apache/shardingsphere/migration/distsql/handler/query/ShowMigrationJobStatusQueryResultSet.java:60]: It's `null != jobItemProgress` in if condition, but it use `jobItemInfo` here, it's not clear enough as before

File: kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/pojo/MigrationJobItemInfo.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package org.apache.shardingsphere.data.pipeline.api.pojo;
+
+import lombok.Getter;
+
+/**
+ * Migration job item info.
+ */
+@Getter
+public final class MigrationJobItemInfo extends InventoryIncrementalJobItemInfo {
```
Comment (sandynz) [inline kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/api/pojo/MigrationJobItemInfo.java:26]: Looks it's not necessary to create `MigrationJobItemInfo`, we could keep them in `InventoryIncrementalJobItemInfo` for common usage

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/migration/MigrationJobAPIImpl.java [hunk 4]
```diff
@@ -133,6 +137,32 @@ protected TableBasedPipelineJobInfo getJobInfo(final String jobId) {
         return new TableBasedPipelineJobInfo(jobMetaData, getJobConfiguration(jobConfigPOJO).getSourceTableName());
     }
     
+    @Override
+    public List<InventoryIncrementalJobItemInfo> getJobItemInfos(final String jobId) {
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/migration/MigrationJobAPIImpl.java:141]: Could keep it in `AbstractInventoryIncrementalJobAPIImpl` and extend it for common usage

**Review by sandynz — CHANGES_REQUESTED** [review]

**Review by sandynz — APPROVED** [review]

**Pull request merged**